core/NameRef.h

```diff
@@ -44,7 +44,7 @@ class NameRef final : private DebugOnlyCheck<NameRefDebugCheck> {
     friend GlobalState;
     friend Name;
 
-    NameRef() : _id(-1){};
+    NameRef() : _id(0){};
```

Comment:
This is the core of the change. There is no distinguished `-1` variable for a NameRef.